src/libmeasurement_kit/common/utils.cpp

```diff
@@ -57,8 +48,8 @@ timeval *timeval_init(timeval *tv, double delta) {
     if (delta < 0) {
         return nullptr;
     }
-    tv->tv_sec = (time_t)floor(delta);
-    tv->tv_usec = (suseconds_t)((delta - floor(delta)) * 1000000);
```

Comment:
These cases were creating portability problems (suseconds_t not defined on Windows and struct timeval has all fields as long), so I decided to sacrifice them to move forward, without losing pretty much anything.